src/pg.cr

```diff
@@ -14,31 +14,41 @@ module PG
   #   pp notification.channel, notification.payload, notification.pid
   # end
   # ```
-  def self.connect_listen(url, *channels : String, &blk : PQ::Notification ->) : ListenConnection
-    connect_listen(url, channels, &blk)
+  #
+  # By default, this will spawn a fiber to non-blocking listen. If you would
+  # rather handle this yourself, pass true to the blocking parameter.
+  #
+  # ```
+  # PG.connect_listen("postgres:///", "a", "b", blocking: true) { ... }
+  # ```
+
+  def self.connect_listen(url, *channels : String, blocking : Bool = false, &blk : PQ::Notification ->) : ListenConnection
```

Comment:
LGTM, yet it is not clear to me that is users responsibility to close the underlying connection from the return value.

Reply:
I think with the blocking case then it's on the user. Before it'd just stay open for the lifetime of the program and there was no real way to close it. But with this, it stays open until they close it.